### ISSUE TITLE
How to build without using standalone toolchain?

### ISSUE THREAD

**Issue by rcdailey** [android-ndk/ndk#773]
It's mentioned in the roadmap for NDK that r19 or r20 (I forget which one) will deprecate or remove standalone toolchains. However, I do not see a guide anywhere today that shows how to build libraries against the NDK toolchains _without_ using the standalone toolchains. There's [a very good writeup](https://developer.android.com/ndk/guides/standalone_toolchain) on how to use the toolchain, but not one for without that I could find.

Basically if I build the Boost library, I can probably point `--sysroot` to the `sysroot` directory at the root, but I'm not sure how to do things like specify the STL to use, or specify the minimum API level. Does such a guide already exist? And if not, could this task serve as a request for such a guide?

**Comment by enh** [comment]
the key point is "deprecate or remove standalone toolchains *by making every toolchain a standalone toolchain*". you shouldn't be doing any prep work yet, because there's no new thing to move to yet. right now, you're doing it right if you're using a standalone toolchain.

r18 will remove stlport and gnustl, so there will no longer be any need to specify an STL. (i'll update the docs to reflect that when r18 ships.)

the API level is the only thing that remains, and that will be part of the triple.

**Comment by enh** [comment]
(i'll leave this open for now, since our wording has definitely caused some confusion.)

**Comment by rcdailey** [comment]
Ah, ok I misunderstood. Thank you for clarifying. I thought that we were expected to somehow use the existing NDK structure as a replacement for standalone, but that's not the case.

**Comment by DanAlbert** [comment]
FYI, I also have no plans of getting rid of the standalone toolchain script. The only difference there is that it will essentially just be an overly complicated `cp`.

**Comment by DanAlbert** [comment]
Dup of https://github.com/android-ndk/ndk/issues/780 (and https://github.com/android-ndk/ndk/issues/125) and which is fixed in r19.

Simplified by https://github.com/android-ndk/ndk/issues/812.

**Comment by markand** [comment]
Sorry to ask here, I also like standalone toolchains because they are just easier to use when building external libraries that are autotools based. Using them you don't have to pass dozens of parameters. So it is safe to continue relying on it for the future?
